posthog/api/user.py

```diff
@@ -39,7 +38,7 @@ class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
         fields = [
-            "date_joined",
+            "id",
```

Comment:
Are we sure this is absolutely needed? Feels like a regression to go back to using the PK `id` instead of the `uuid`

Reply:
Hm, well, most important for me is consistency, in that wherever we use users as keys in the API, we should stick to a single column. Currently organization members use `id`, so I went for that in project memberships too. But I can close this and instead do it the other way around: move organization members to `uuid` in a separate PR, and use `uuid` too in #5894.

Reply:
Alternative PR: #5971